blackjax/mcmc/proposal.py

```diff
@@ -16,8 +16,23 @@
 import jax
 import jax.numpy as jnp
 import numpy as np
+from typing_extensions import Protocol
 
-from blackjax.mcmc.integrators import IntegratorState
+from blackjax.types import PyTree
+
+
+class StateWithPosition(Protocol):
```

Comment:
Check that StateWithPosition is a Protocol. By having it, we can have Interface Segretation Principle on proposal_generator_from_energy and there's no need to create an IntegratorState for RMH since RWSTate satisfies the protocol.